source/host_utils.c

```diff
@@ -79,24 +76,38 @@ bool aws_host_utils_is_ipv6(struct aws_byte_cursor host, bool is_uri_encoded) {
     bool is_split = aws_byte_cursor_next_split(&host, '%', &substr);
     AWS_ASSERT(is_split); /* function is guaranteed to return at least one split */
 
-    if (!is_split || substr.len == 0 || s_ends_with(substr, ':') ||
-        !aws_byte_cursor_satisfies_pred(&substr, s_is_ipv6_char)) {
+    if (!is_split || substr.len < 2 || substr.len > 39 || !aws_byte_cursor_satisfies_pred(&substr, s_is_ipv6_char)) {
+        return false;
+    }
+
+    if ((substr.ptr[0] == ':' && substr.ptr[1] != ':') ||                           /* no single colon at start */
+        (substr.ptr[substr.len - 1] == ':' && substr.ptr[substr.len - 2] != ':')) { /* no single colon at end */
```

Comment:
is there any RFC/spec we can link to for the rules of ipv6?
I only find [this](https://www.ibm.com/docs/en/i/7.5?topic=concepts-ipv6-address-formats) from IBM?